src/Data/Nat/Divisibility.agda

```diff
@@ -25,24 +25,68 @@ open import Relation.Binary.Definitions
   using (Reflexive; Transitive; Antisymmetric; Decidable)
 import Relation.Binary.Reasoning.Preorder as PreorderReasoning
 open import Relation.Binary.PropositionalEquality.Core
-  using (_≡_; _≢_; refl; sym; trans; cong; cong₂; subst)
+  using (_≡_; _≢_; refl; sym; cong; subst)
 open import Relation.Binary.Reasoning.Syntax
-import Relation.Binary.PropositionalEquality.Properties as PropEq
+open import Relation.Binary.PropositionalEquality.Properties
+  using (isEquivalence; module ≡-Reasoning)
+
+private
+  variable d m n o p : ℕ
+
+
+------------------------------------------------------------------------
+-- Definition and derived properties
+
+open import Data.Nat.Divisibility.Core public hiding (*-pres-∣)
+
+module _ (m∣n : m ∣ n) where
+
+  open _∣_ m∣n
+  open ≤-Reasoning
+
+  instance
+    quotient≢0 : .{{NonZero n}} → NonZero quotient
+    quotient≢0 rewrite equality = m*n≢0⇒m≢0 quotient
+
+  equalityᵒ : n ≡ m * quotient
```

Comment:
What does the `ᵒ` in the name stand for? I'm not a huge fan of this name to be honest. Maybe `quotient-≡` would be more in keeping with the other names?

Reply:
Okay I think the actual problem is that `equality` is a frankly terrible name. How about we define a more meaningful alias name for `equality` in the record `m|n=>n≡ quotient*m` and then this can be ``m|n=>n≡ m*quotient`?

Reply:
Agree that `equality` in the record is a terrible name!
I think I didn't understand: where should the proposed alias live? As a manifest field in the record?

Reply:
I mean I would personally put it in the section with all the other stuff you moved out of the record... but I realise that may make your DRY objection worse, so happy to wait for the resolution of that conversation before resolving this one!

Reply:
Well, in a way the issue is orthogonal to the DRY one, so I've pushed the alias definition as is.